OpenEdXMobile/src/main/java/org/edx/mobile/util/InAppPurchasesUtils.kt

```diff
@@ -11,10 +12,13 @@ object InAppPurchasesUtils {
      * through google play store.
      * */
     fun getInCompletePurchases(
-        auditCoursesSku: List<String>,
+        auditCourses: List<AuditCourse>,
         purchases: List<Pair<String, String>>
-    ): MutableList<Pair<String, String>> {
-        return purchases.filter { it.first in auditCoursesSku }.toMutableList()
+    ): MutableList<AuditCourse> {
+        auditCourses.forEach { courses ->
+            purchases.find { it.first == courses.id }?.let { courses.purchaseToken = it.second }
+        }
+        return auditCourses.filter { it.purchaseToken.isNullOrBlank().not() }.toMutableList()
```

Comment:
this logic needs to be checked, cuz using this method app is unable to identify the unfulfilled courses.